### PR TITLE
Store and use Conversation Participant

### PR DESCRIPTION
**Pull request by PopDaph**
Model to store the list of participants who did something on a given conversation.

### CODE REVIEW

**Comment by spolu** [comment]
Model LGTM

**Comment by PopDaph** [comment]
@spolu I added a commit to use this new model directly in this PR as it's fairly small 🙏🏻

File: front/lib/api/assistant/conversation.ts [hunk 1]
```diff
@@ -223,16 +221,9 @@ export async function getUserConversations(
     attributes: ["id", "sId", "title", "createdAt"],
     include: [
       {
-        model: Message,
+        model: ConversationParticipant,
+        where: { userId: user.id, action: "posted" },
```
Comment (spolu) [inline front/lib/api/assistant/conversation.ts:225]: Shouldn't this be done the other way around? 

If not, is this correclty optimized?

File: front/lib/api/assistant/conversation.ts [hunk 2]
```diff
@@ -499,6 +490,17 @@ export async function* postUserMessage(
         context: context,
       };
 
+      if (user) {
```
Comment (spolu) [inline front/lib/api/assistant/conversation.ts:493]: Should be done concurrently to the Message.create call above?

Also should be created only if it does not exist?

File: front/lib/models/assistant/conversation.ts [hunk 3]
```diff
@@ -86,6 +87,61 @@ Conversation.belongsTo(Workspace, {
...
+    },
+    action: {
+      type: DataTypes.STRING,
+      allowNull: false,
+    },
+  },
+  {
+    modelName: "conversation_participant",
+    sequelize: front_sequelize,
+  }
```
Comment (spolu) [inline front/lib/models/assistant/conversation.ts:128]: We probably want a unique index on conversationId userId and an index on userId ?

File: front/lib/api/assistant/conversation.ts (hunk 2 above)
Comment (PopDaph) [reply front/lib/api/assistant/conversation.ts:493]: If we want to use it at some point for a notification system we'll have to have the last date when an action occurred, but at the moment indeed it makes way more sense to create only if not exist. Adding a commit!

File: front/lib/models/assistant/conversation.ts (hunk 3 above)
Comment (PopDaph) [reply front/lib/models/assistant/conversation.ts:136]: I added indexes for both userId, and (userId, conversationId). 
Adding the unique would mean we won't be able to store multiple actions for the same couple user/convo, do we want that?

Comment (spolu) [reply front/lib/models/assistant/conversation.ts:136]: I thiiink if it's called "Participant" ?

**Review by spolu — COMMENTED** [review]
LGTM

as per IRL I think we can keep the message creation logic in one promise and all the participant logic in another and then call promise.all on both so that we have for both 2 round trips to DB in parallel vs here

1 round trip for participant
1 round trip for UserMessage
and then
1 and 1 round trip concurrent

**Comment by spolu** [comment]
aka 3 round trip sequentially vs 2

**Review by spolu — APPROVED** [review]

**Pull request merged**